ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/CommandFactories.java

```diff
@@ -128,223 +108,40 @@ private CreateStreamCommand handleCreateStream(
       final CallInfo callInfo,
       final CreateStream statement
   ) {
-    final SourceName sourceName = statement.getName();
-    final KsqlTopic topic = buildTopic(statement.getProperties(), serviceContext);
-    final LogicalSchema schema = buildSchema(statement.getElements());
-    final Optional<ColumnName> keyFieldName = buildKeyFieldName(statement, schema);
-    final TimestampExtractionPolicy timestampExtractionPolicy = buildTimestampExtractor(
-        callInfo.ksqlConfig,
-        statement.getProperties(),
-        schema
-    );
-    final Set<SerdeOption> serdeOptions = serdeOptionsSupplier.build(
-        schema,
-        topic.getValueFormat().getFormat(),
-        statement.getProperties().getWrapSingleValues(),
-        callInfo.ksqlConfig
-    );
-    validateSerdeCanHandleSchemas(
-        callInfo.ksqlConfig,
-        serviceContext,
-        serdeFactory,
-        PhysicalSchema.from(schema, serdeOptions),
-        topic
-    );
-    return new CreateStreamCommand(
+    return createSourceFactory.createStreamCommand(
         callInfo.sqlExpression,
-        sourceName,
-        schema,
-        keyFieldName,
-        timestampExtractionPolicy,
-        serdeOptions,
-        topic
+        statement,
+        callInfo.ksqlConfig
     );
   }
 
   private CreateTableCommand handleCreateTable(
       final CallInfo callInfo,
       final CreateTable statement
   ) {
-    final SourceName sourceName = statement.getName();
-    final KsqlTopic topic = buildTopic(statement.getProperties(), serviceContext);
-    final LogicalSchema schema = buildSchema(statement.getElements());
-    final Optional<ColumnName> keyFieldName = buildKeyFieldName(statement, schema);
-    final TimestampExtractionPolicy timestampExtractionPolicy = buildTimestampExtractor(
-        callInfo.ksqlConfig,
-        statement.getProperties(),
-        schema
-    );
-    final Set<SerdeOption> serdeOptions = serdeOptionsSupplier.build(
-        schema,
-        topic.getValueFormat().getFormat(),
-        statement.getProperties().getWrapSingleValues(),
-        callInfo.ksqlConfig
-    );
-    validateSerdeCanHandleSchemas(
-        callInfo.ksqlConfig,
-        serviceContext,
-        serdeFactory,
-        PhysicalSchema.from(schema, serdeOptions),
-        topic
-    );
-    return new CreateTableCommand(
+    return createSourceFactory.createTableCommand(
         callInfo.sqlExpression,
-        sourceName,
-        schema,
-        keyFieldName,
-        timestampExtractionPolicy,
-        serdeOptions,
-        topic
+        statement,
+        callInfo.ksqlConfig
     );
   }
 
   private DropSourceCommand handleDropStream(final DropStream statement) {
-    return handleDropSource(
-        statement.getName(),
-        statement.getIfExists(),
-        DataSourceType.KSTREAM
-    );
+    return dropSourceFactory.create(statement);
```

Comment:
Just wondering if you could not inline each of these methods in the static map??